src/components/Button/src/Button.vue

```diff
@@ -130,37 +144,110 @@ export default {
 			default: undefined,
 		},
 		/**
-		 * Background color of button
+		 * Main color of button
 		 */
 		color: {
 			type: String,
 			default: undefined,
 			validator: (color) => colord(color).isValid(),
 		},
 		/**
-		 * Text color of button
+		 * Main hover color of button
+		 */
+		colorHover: {
+			type: String,
+			default: undefined,
+			validator: (color) => colord(color).isValid(),
+		},
+		/**
+		 * Text color of button (only applied on fill buttons)
 		 */
 		textColor: {
 			type: String,
 			default: undefined,
 			validator: (color) => colord(color).isValid(),
 		},
+		/**
+		 * Text hover color of button (only applied on fill buttons)
+		 */
+		textColorHover: {
+			type: String,
+			default: undefined,
+			validator: (color) => colord(color).isValid(),
+		},
 		/**
 		 * Variant
-		 * @values primary, secondary, tertiary
+		 * @values fill, outline, ghost
 		 */
 		variant: {
 			type: String,
 			default: undefined,
-			validator: (variant) => ['primary', 'secondary', 'tertiary'].includes(variant),
+			validator: (variant) => ['fill', 'outline', 'ghost'].includes(variant),
 		},
 		/**
 		 * Shape of button
+		 * @values squared, rounded, pill, 'Npx', 'N%'
 		 */
 		shape: {
 			type: String,
 			default: undefined,
-			validator: (shape) => ['squared', 'rounded', 'pill'].includes(shape),
+			validator: (shape) => ['squared', 'rounded', 'pill'].includes(shape) || shape.match(/\d*\.?\d+(?:px|%)/i),
```

Comment:
hmmm, did you consider making a `borderRadius` prop? e.g.
```js
borderRadius: {
  type: String,
  default: undefined,
  validator: cssValidator('border-radius'),
}
```
and if a `borderRadius` is passed we can ignore the `shape` prop. It would allow much greater customization, the current `shape` prop and its validator wouldn't allow people to customize the border-radius of the individual button corners independently, so things like "wavy" buttons wouldn't be possible, e.g. https://codepen.io/kfedorov91/pen/gOzemvN

I'm not sure if we want that level of customizability buuuut if we do it might just be worth it to add a `borderRadius` prop if it's possible and makes sense.